File: src/lessons/recursion/cons/NthEntity.py
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(nth(toRecListIfArray(param[0]), toRecListIfArray(param[1]))))

# BEGIN TEMPLATE
def nth(list, n):
# BEGIN SOLUTION
  if n == 1:
    return list.head
  return nth(list.tail, n-1)
# END SOLUTION
# END TEMPLATE

File: src/lessons/recursion/cons/NthEntity.java
package lessons.recursion.cons;

import static plm.core.ValueSerializer.*;

import lessons.recursion.cons.universe.ConsEntity;
import lessons.recursion.cons.universe.RecList;

public class NthEntity extends ConsEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(nth(RecList.fromArray(toPrimitive(toArrayOfType(param[0], Integer.class))), (Integer)param[1])));
    }
  }

  /* BEGIN TEMPLATE */
  int nth(RecList seq, int n)
  {
    /* BEGIN SOLUTION */
    if (n == 1)
      return seq.head;
    return nth(seq.tail, n - 1);
    /* END SOLUTION */
  }
  /* END TEMPLATE */
}
